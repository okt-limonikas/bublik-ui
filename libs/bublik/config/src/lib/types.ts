/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { z } from 'zod';

export const BublikEnvSchema = z.object({
	BASE_URL: z.string(),
	MODE: z.string(),
	DEV: z.boolean(),
	PROD: z.boolean(),
	SSR: z.boolean()
});

export const BublikConfigSchema = z.object({
	baseUrl: z.string(),
	baseUrlApi: z.string(),
	oldBaseUrl: z.string().optional(),
	rootUrl: z.string(),
	dashboardPollingTimer: z.number(),
	isProd: z.boolean(),
	isDev: z.boolean(),
	isSsr: z.boolean(),
	mode: z.string()
});

export type BublikEnv = z.infer<typeof BublikEnvSchema>;

export type BublikConfig = z.infer<typeof BublikConfigSchema>;
