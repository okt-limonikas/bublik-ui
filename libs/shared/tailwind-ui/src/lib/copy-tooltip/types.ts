/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
export interface CopyProps {
	copyString?: string;
	onCopyComplete?: (isSuccess: boolean) => void;
}

export type CopyContentHandle = {
	resetState: () => void;
};
