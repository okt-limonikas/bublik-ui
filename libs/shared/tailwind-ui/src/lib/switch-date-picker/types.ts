import { DateValue } from '@internationalized/date';

import { RangeValue } from './utils';

export type DateRange = {
	label: string;
	range: RangeValue<DateValue>;
};
