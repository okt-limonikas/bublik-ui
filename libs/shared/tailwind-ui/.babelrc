{
	"presets": [
		[
			"@nx/react/babel",
			{
				"runtime": "automatic",
				"useBuiltIns": "usage"
			}
		],
		"@emotion/babel-preset-css-prop"
	],
	"plugins": []
}
