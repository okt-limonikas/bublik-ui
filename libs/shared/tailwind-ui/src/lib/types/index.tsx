/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
export type AsProp<C extends React.ElementType> = {
	as?: C;
};

export type PropsToOmit<C extends React.ElementType, P> = keyof (AsProp<C> & P);

export type PolymorphicComponentProp<
	C extends React.ElementType,
	Props = Record<string, unknown>
> = React.PropsWithChildren<Props & AsProp<C>> &
	Omit<React.ComponentPropsWithoutRef<C>, PropsToOmit<C, Props>>;

export type PolymorphicComponentPropWithRef<
	C extends React.ElementType,
	Props = Record<string, unknown>
> = PolymorphicComponentProp<C, Props> & { ref?: PolymorphicRef<C> };

export type PolymorphicRef<C extends React.ElementType> =
	React.ComponentPropsWithRef<C>['ref'];
