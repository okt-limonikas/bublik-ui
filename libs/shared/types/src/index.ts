/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
export * from './lib/dashboard';
export * from './lib/tree';
export * from './lib/history';
export * from './lib/log';
export * from './lib/runs';
export * from './lib/run';
export * from './lib/deploy-info';
export * from './lib/measurements';
export * from './lib/run-import';
export * from './lib/log-json-schema';
export * from './lib/auth';
export * from './lib/import';
export * from './lib/performance';
export * from './lib/report';
