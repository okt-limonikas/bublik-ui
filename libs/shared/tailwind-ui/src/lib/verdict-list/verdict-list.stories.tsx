/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { Meta } from '@storybook/react';

import { RESULT_TYPE } from '@/shared/types';

import { VerdictList } from './verdict-list';

import { withBackground } from '../storybook-bg';

export default {
	component: VerdictList,
	title: 'components/Verdict List',
	decorators: [withBackground]
} as Meta;

export const Obtained = {
	args: {
		variant: 'obtained',
		result: RESULT_TYPE.Passed,
		verdicts: [
			'verdict-1',
			'verdict-2',
			'verdict-3',
			'verdict-4',
			'verdict-5',
			'verdict-6'
		],
		isResultSelected: false,
		selectedVerdicts: ['verdict-5', 'verdict-3']
	}
};

export const Expected = {
	args: {
		variant: 'expected',
		result: RESULT_TYPE.Passed,
		verdicts: [
			'verdict-1',
			'verdict-2',
			'verdict-3',
			'verdict-4',
			'verdict-5',
			'verdict-6'
		],
		isResultSelected: false,
		selectedVerdicts: ['verdict-5', 'verdict-3']
	}
};
