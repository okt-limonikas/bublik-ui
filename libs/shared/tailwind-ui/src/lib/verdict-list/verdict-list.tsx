/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import React, { FC } from 'react';

import { RESULT_TYPE } from '@/shared/types';

import { Badge, BadgeVariants } from '../badge';

export const enum VerdictVariant {
	Expected = 'expected',
	Obtained = 'obtained'
}

export type VerdictVariantValue = 'expected' | 'obtained';

export interface VerdictResultProps {
	variant: VerdictVariant | VerdictVariantValue;
	resultType: RESULT_TYPE;
	isNotExpected?: boolean;
	isSelected?: boolean;
	onResultClick?: (resultType: RESULT_TYPE) => void;
}

export const VerdictResult = (props: VerdictResultProps) => {
	const { variant, resultType, onResultClick, isNotExpected, isSelected } =
		props;

	const resultVariant =
		variant === 'obtained'
			? isNotExpected
				? BadgeVariants.Unexpected
				: BadgeVariants.Expected
			: BadgeVariants.Transparent;

	return (
		<Badge
			variant={resultVariant}
			isSelected={isSelected}
			onClick={onResultClick ? () => onResultClick?.(resultType) : undefined}
		>
			{resultType}
		</Badge>
	);
};

export interface ListOfVerdictProps {
	verdicts: VerdictListProps['verdicts'];
	selectedVerdicts: VerdictListProps['selectedVerdicts'];
	onVerdictClick: VerdictListProps['onVerdictClick'];
}

export const ListOfVerdicts: FC<ListOfVerdictProps> = ({
	verdicts,
	selectedVerdicts,
	onVerdictClick
}) => {
	return (
		<div className="flex flex-col flex-wrap gap-1">
			{verdicts.map((verdict) => {
				const isSelected = selectedVerdicts?.includes(verdict);

				return (
					<Badge
						key={verdict}
						variant={BadgeVariants.Transparent}
						isSelected={isSelected}
						onClick={onVerdictClick ? () => onVerdictClick(verdict) : undefined}
						className="text-start"
						overflowWrap
					>
						{verdict}
					</Badge>
				);
			})}
		</div>
	);
};

export interface VerdictListProps {
	variant: VerdictVariant | VerdictVariantValue;
	result: RESULT_TYPE;
	verdicts: string[];
	isResultSelected?: boolean;
	isNotExpected?: boolean;
	selectedVerdicts?: string[];
	onResultClick?: (resultValue: RESULT_TYPE) => void;
	onVerdictClick?: (verdict: string) => void;
}

export const VerdictList: FC<VerdictListProps> = (props) => {
	const {
		variant,
		result,
		verdicts,
		isResultSelected,
		selectedVerdicts,
		onResultClick,
		onVerdictClick,
		isNotExpected
	} = props;

	return (
		<div className="flex flex-col gap-1" data-testid="tw-verdict-list">
			<VerdictResult
				variant={variant}
				resultType={result}
				isNotExpected={isNotExpected}
				isSelected={isResultSelected}
				onResultClick={onResultClick}
			/>
			<ListOfVerdicts
				verdicts={verdicts}
				selectedVerdicts={selectedVerdicts}
				onVerdictClick={onVerdictClick}
			/>
		</div>
	);
};
