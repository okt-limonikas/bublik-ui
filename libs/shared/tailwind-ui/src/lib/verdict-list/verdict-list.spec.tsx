/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { it, describe, expect } from 'vitest';

import { render } from '@testing-library/react';

import { RESULT_TYPE } from '@/shared/types';

import { VerdictList, VerdictListProps } from './verdict-list';

const getDefaultProps = (): VerdictListProps => {
	return {
		verdicts: ['verdict-1', 'verdict-2', 'verdict-3'],
		variant: 'obtained',
		result: RESULT_TYPE.Passed
	};
};

describe('VerdictList', () => {
	it('should render successfully', () => {
		const { baseElement } = render(<VerdictList {...getDefaultProps()} />);

		expect(baseElement).toBeTruthy();
	});

	it('should match snapshot', () => {
		const { asFragment } = render(<VerdictList {...getDefaultProps()} />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should highlight selected verdicts', () => {
		const { asFragment } = render(
			<VerdictList
				{...getDefaultProps()}
				isNotExpected={true}
				selectedVerdicts={['verdict-1']}
				variant="expected"
			/>
		);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should render expected result (TRANSPARENT)', () => {
		const { asFragment } = render(
			<VerdictList
				{...getDefaultProps()}
				isNotExpected={true}
				variant="expected"
			/>
		);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should render not expected result (RED)', () => {
		const { asFragment } = render(
			<VerdictList
				{...getDefaultProps()}
				isNotExpected={true}
				variant="obtained"
			/>
		);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should render not expected result (GREEN)', () => {
		const { asFragment } = render(
			<VerdictList
				{...getDefaultProps()}
				isNotExpected={false}
				variant="obtained"
			/>
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
