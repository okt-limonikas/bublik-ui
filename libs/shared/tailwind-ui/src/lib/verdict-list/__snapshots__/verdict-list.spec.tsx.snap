// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VerdictList > should highlight selected verdicts 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-1"
    data-testid="tw-verdict-list"
  >
    <div
      class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary"
      data-testid="tw-badge"
    >
      PASSED
    </div>
    <div
      class="flex flex-col flex-wrap gap-1"
    >
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors overflow-wrap-anywhere text-start bg-primary-wash border-primary"
        data-testid="tw-badge"
      >
        verdict-1
      </div>
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary overflow-wrap-anywhere text-start"
        data-testid="tw-badge"
      >
        verdict-2
      </div>
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary overflow-wrap-anywhere text-start"
        data-testid="tw-badge"
      >
        verdict-3
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`VerdictList > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-1"
    data-testid="tw-verdict-list"
  >
    <div
      class="inline-flex items-center w-fit py-0.5 px-2 rounded border border-transparent text-[0.75rem] font-medium transition-colors text-text-expected bg-badge-3"
      data-testid="tw-badge"
    >
      PASSED
    </div>
    <div
      class="flex flex-col flex-wrap gap-1"
    >
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary overflow-wrap-anywhere text-start"
        data-testid="tw-badge"
      >
        verdict-1
      </div>
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary overflow-wrap-anywhere text-start"
        data-testid="tw-badge"
      >
        verdict-2
      </div>
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary overflow-wrap-anywhere text-start"
        data-testid="tw-badge"
      >
        verdict-3
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`VerdictList > should render expected result (TRANSPARENT) 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-1"
    data-testid="tw-verdict-list"
  >
    <div
      class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary"
      data-testid="tw-badge"
    >
      PASSED
    </div>
    <div
      class="flex flex-col flex-wrap gap-1"
    >
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary overflow-wrap-anywhere text-start"
        data-testid="tw-badge"
      >
        verdict-1
      </div>
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary overflow-wrap-anywhere text-start"
        data-testid="tw-badge"
      >
        verdict-2
      </div>
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary overflow-wrap-anywhere text-start"
        data-testid="tw-badge"
      >
        verdict-3
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`VerdictList > should render not expected result (GREEN) 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-1"
    data-testid="tw-verdict-list"
  >
    <div
      class="inline-flex items-center w-fit py-0.5 px-2 rounded border border-transparent text-[0.75rem] font-medium transition-colors text-text-expected bg-badge-3"
      data-testid="tw-badge"
    >
      PASSED
    </div>
    <div
      class="flex flex-col flex-wrap gap-1"
    >
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary overflow-wrap-anywhere text-start"
        data-testid="tw-badge"
      >
        verdict-1
      </div>
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary overflow-wrap-anywhere text-start"
        data-testid="tw-badge"
      >
        verdict-2
      </div>
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary overflow-wrap-anywhere text-start"
        data-testid="tw-badge"
      >
        verdict-3
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`VerdictList > should render not expected result (RED) 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-1"
    data-testid="tw-verdict-list"
  >
    <div
      class="inline-flex items-center w-fit py-0.5 px-2 rounded border border-transparent text-[0.75rem] font-medium transition-colors bg-bg-fillError text-text-unexpected"
      data-testid="tw-badge"
    >
      PASSED
    </div>
    <div
      class="flex flex-col flex-wrap gap-1"
    >
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary overflow-wrap-anywhere text-start"
        data-testid="tw-badge"
      >
        verdict-1
      </div>
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary overflow-wrap-anywhere text-start"
        data-testid="tw-badge"
      >
        verdict-2
      </div>
      <div
        class="inline-flex items-center w-fit py-0.5 px-2 rounded border text-[0.75rem] font-medium transition-colors bg-transparent border-border-primary overflow-wrap-anywhere text-start"
        data-testid="tw-badge"
      >
        verdict-3
      </div>
    </div>
  </div>
</DocumentFragment>
`;
